Core/Core/CourseSync/Common/Models/CourseSyncListerInteractor.swift

```diff
@@ -0,0 +1,177 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2023-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Combine
+import CombineSchedulers
+import Foundation
+
+protocol CourseSyncListInteractor {
+    func getCourseSyncEntries(filter: CourseSyncListInteractorLive.Filter) -> AnyPublisher<[CourseSyncEntry], Error>
```

Comment:
I used to extract such enums into a separate entity. It doesn't seem to be correct that a protocol definition uses a type from a concrete implementation of the protocol.

Reply:
I agree, was a bit hesitant but ended up leaving it there. I moved it to a separate entity and a separate file.

Reply:
It would be nice if we could embed an enum into a protocol.